### PR TITLE
RDK-35648: Override HDMI Color Depth

### PR DESCRIPTION
**Pull request by acheri988**
Reason for change:
Override HDMI Color Depth
Test Procedure: None
Risks: Low

Change-Id: I31cedb96ed00345ac0e8916c2c05e65790b42c2a
Signed-off-by:Anooj Cheriyan <Anooj_Cheriyan@comcast.com>

### CODE REVIEW

**Comment by Firdose2109** [comment]
Can one of the admins verify this patch?

File: DisplaySettings/DisplaySettings.cpp [hunk 1]
```diff
@@ -3680,6 +3684,109 @@ namespace WPEFramework {
             returnResponse(success);
         }
 
+        uint32_t DisplaySettings::getPreferredColorDepth(const JsonObject& parameters, JsonObject& response)
+        {   //sample servicemanager response:{"colorDepth":"10 Bit","success":true}
+            LOGINFOMETHOD();
+            std::string strVideoPort = device::Host::getInstance().getDefaultVideoPortName();
+            string videoDisplay = parameters.HasLabel("videoDisplay") ? parameters["videoDisplay"].String() : strVideoPort;
+            bool hasPersist = parameters.HasLabel("persist");
```
Comment (apatel859) [inline DisplaySettings/DisplaySettings.cpp:3692]: remove has persist
bool persist = parameters.HasLabel("persist") ? parameters["persist"].Boolean() : true;

File: DisplaySettings/DisplaySettings.cpp [hunk 2]
```diff
@@ -3680,6 +3684,109 @@ namespace WPEFramework {
...
+        {   //sample servicemanager response: {"success":true}
+            LOGINFOMETHOD();
+            returnIfParamNotFound(parameters, "videoDisplay");
+            returnIfParamNotFound(parameters, "colorDepth");
+
+            string videoDisplay = parameters["videoDisplay"].String();
+            string strColorDepth = parameters["colorDepth"].String();
+
+            bool hasPersist = parameters.HasLabel("persist");
+            bool persist = hasPersist ? parameters["persist"].Boolean() : true;
```
Comment (apatel859) [inline DisplaySettings/DisplaySettings.cpp:3737]: same as above

File: DisplaySettings/DisplaySettings.cpp [hunk 3]
```diff
@@ -3680,6 +3684,109 @@ namespace WPEFramework {
...
+            if (!hasPersist) LOGINFO("persist: true");
+
+            bool success = true;
+            try
+            {
+                device::VideoOutputPort &vPort = device::Host::getInstance().getVideoOutputPort(videoDisplay);
+		if (0==strcmp(strColorDepth.c_str(), "8 Bit")){
+			vPort.setPreferredColorDepth(dsDISPLAY_COLORDEPTH_8BIT, persist);
+		} else if (0==strcmp(strColorDepth.c_str(), "10 Bit")){
```
Comment (apatel859) [inline DisplaySettings/DisplaySettings.cpp:3746]: strncmp

File: DisplaySettings/DisplaySettings.h [hunk 4]
```diff
@@ -148,6 +148,11 @@ namespace WPEFramework {
             uint32_t resetVolumeLeveller(const JsonObject& parameters, JsonObject& response);
             uint32_t getVideoFormat(const JsonObject& parameters, JsonObject& response);
             uint32_t setMS12ProfileSettingsOverride(const JsonObject& parameters, JsonObject& response);
+
+            uint32_t setPreferredColorDepth(const JsonObject& parameters, JsonObject& response);
+            uint32_t getPreferredColorDepth(const JsonObject& parameters, JsonObject& response);
+            uint32_t getColorDepthCapabilities(const JsonObject& parameters, JsonObject& response);
+
             void InitAudioPorts();
```
Comment (apatel859) [inline DisplaySettings/DisplaySettings.h:156]: update json file for documenting this 3 API

**Comment by acheri988** [comment]
New PR created: 2519

**Pull request closed without merging**